Automattic/pocket-casts-android
# Adding more refresh logging and converting code to Coroutines

## Description

This change started by adding extra logging to help debug slow refreshing. I was finding it hard to add these timing logs so I converted some of the code from RxJava to Coroutines. The main method converted is `PodcastManager.refreshPodcastInBackground`. This called quite a few blocking methods so I have added alternative methods with the suffix `Suspend` until we can remove them. 

## Testing Instructions

I have added a new test for the rewrite but maybe just opening the podcast page and also running a refresh to see the new timings appear.

```
BgTask: Refresh - sync update - 385 ms
BgTask: Refresh - process server response - 1 ms
```

## Checklist
 
#### I have tested any UI changes...
